src/c++/perf_analyzer/perf_analyzer.cc

```diff
@@ -52,6 +52,44 @@ SignalHandler(int signum)
     exit(0);
   }
 }
+
+bool
+IsLLMModel(
+    const std::shared_ptr<pa::ModelParser>& parser,
+    const pa::PAParamsPTR& params)
+{
+  bool is_llm_from_user = params->is_llm if (is_llm_from_user)
+  {
+    return true;
+  }
+
+  bool is_llm = false;
+  // check if its decoupled
+  is_llm =
+      is_llm || (parser->IsDecoupled() && !params->profile_export_file.empty());
+
+  // check if is ensemble model, and if model has a tensorrt_llm portion to it
+  // then it is for sure the tensorrt-llm backend
+  if (!parser->composing_models_map_.empty()) {
+    auto composing_models_map = parser->composing_models_map_;
+    for (auto& [_, model_version_pair] : *composing_models_map) {
```

Comment:
I am starting to suspect that my c++ is out of date. 
You can do tuple deconstruction like this?

Reply:
Seems this is a c++ 17 feature. 
Dope. Thanks for the lesson!